src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -399,6 +399,54 @@ inline std::istream& operator>>(std::istream& is, SchedulingCoreType& core_type)
  */
 static constexpr Property<SchedulingCoreType> scheduling_core_type{"SCHEDULING_CORE_TYPE"};
 
+enum class ModelDistributionPolicy {
+    NONE = 0,             // Run one model on single socket/device without parallelism.
+    TENSOR_PARALLEL = 1,  // Split one node or subgraph into parts and run one part per socket/device in parallel.
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const ModelDistributionPolicy& stream_mode) {
+    switch (stream_mode) {
+    case ModelDistributionPolicy::NONE:
+        return os << "NONE";
+    case ModelDistributionPolicy::TENSOR_PARALLEL:
+        return os << "TENSOR_PARALLEL";
+    default:
+        OPENVINO_THROW("Unsupported model distribution policy!");
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, ModelDistributionPolicy& stream_mode) {
+    std::string str;
+    is >> str;
+    if (str == "NONE") {
+        stream_mode = ModelDistributionPolicy::NONE;
+    } else if (str == "TENSOR_PARALLEL") {
+        stream_mode = ModelDistributionPolicy::TENSOR_PARALLEL;
+    } else {
+        OPENVINO_THROW("Unsupported model distribution policy: ", str);
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief This property defines model distribution policy for inference with multiple sockets/devices.
+ * @ingroup ov_runtime_cpp_prop_api
+ *
+ * Developer can use this property to select model distribution policy for CPU inference with multiple sockets
+ * platform or GPU inference with multiple GPU devices.
+ * -- TENSOR_PARALLEL : Split one node or subgraph into parts and run one part per socket/device in parallel.
+ * -- NONE            : Run one model on single socket/device without parallelism.
+ *
+ * The following code is an example to split node into two parts run one part per socket on dual sockets platform.
+ *
+ * @code
+ * ie.set_property(ov::hint::model_distribution_policy(ov::hint::ModelDistributionPolicy::TENSOR_PARTITION));
+ * @endcode
+ */
+static constexpr Property<ModelDistributionPolicy> model_distribution_policy{"MODEL_DISTRIBUTION_POLICY"};
```

Comment:
As I mentioned we should use container instead of pure enum class as a type to allow combination of different policies